### ISSUE TITLE
Move to all `Turing` models in `EpiAware` to be non-vectorised

### ISSUE THREAD

**Issue by SamuelBrand1** [CDCgov/Rt-without-renewal#307]
The problem with a generic forecast function for `EpiAware` #243 is that in general the `Turing.forecast` function doesn't interface smoothly with vectorised random variable sampling (see this discussion https://github.com/TuringLang/Turing.jl/issues/2239).

In various f2f discussions (including with @dylanhmorris ) its been noted that these issues could (probably; we'd need to do some dev work) be resolved by going to a non-vectorised approach (e.g. `filldist` with `Normal` r.v.s rather than `MvNormal`). The downside is the potential performance hit (e.g. https://turinglang.org/docs/tutorials/docs-13-using-turing-performance-tips/ ).

If we want to do this on the way to resolving #243. I think a good step would be to first resolve #300 so we have an accurate before and after picture of performance.

**Comment by seabbs** [comment]
>  its been noted that these issues could (probably; we'd need to do some dev work) be resolved by going to a non-vectorised approach

I agree this seems like it should be better supported with current functionality. 

I also agree this is blocked by benchmarking which I will work on asap so we can unstick this

**Comment by SamuelBrand1** [comment]
Is this unblocked now?

**Comment by seabbs** [comment]
I think we want the full benchmarking and integration profiling I am doing tomorrow in place first
